src/Tribe/Bar.php

```diff
@@ -37,7 +37,7 @@ public function __construct() {
 	 *
 	 */
 	public function should_show() {
-		global $wp_query;
+		$wp_query = tribe_get_global_query_object();
```

Comment:
Pretty much as @mitogh noted, we probably want each of these to be something like:

```php
if ( ! $wp_query = tribe_get_global_query_object() ) {
    return;
}
```